pytorch_lightning/trainer/trainer.py

```diff
@@ -532,6 +532,10 @@ def __init__(
         self.root_gpu = determine_root_gpu_device(self.data_parallel_device_ids)
         self.root_device = torch.device("cpu")
 
+        # self.data_parallel_device_ids is None if gpus is callable
```

Comment:
i don't understand what this means

Reply:
```
self.data_parallel_device_ids = _parse_gpu_ids(self.gpus)
```
_parse_gpu_ids returns None is self.gpus is callable, if you don't pass --gpus flag then self.gpus points to Trainer._arg_default

Reply:
We could move the `self.on_gpu = True if (gpus and torch.cuda.is_available()) else False` found a few lines above
down here and simply write

`self.on_gpu = True if (self.data_parallel_device_ids and torch.cuda.is_available()) else False`

then there is no need for an extra if clause or a comment :)